app/controllers/resources_controller.rb

```diff
@@ -11,6 +11,13 @@ def index
     options = params.permit(*allowed_params)
       .slice(*allowed_params).to_h.symbolize_keys
 
+    if Rails.application.config.conjur_config.api_resource_list_limit.positive?
+      options[:limit] = Rails.application.config.conjur_config.api_resource_list_limit.to_s unless options[:limit]
+
+      raise ApplicationController::UnprocessableEntity, "Limit value should not exceed #{Rails.application.config.conjur_config.api_resource_list_limit}" \
+      unless options[:limit].to_i <= Rails.application.config.conjur_config.api_resource_list_limit
+    end
+
```

Comment:
Generally, I would suggest creating a helper method in the controller (or one of the base classes):

```ruby
def conjur_config
    Rails.application.config.conjur_config
end
```
I don't love it, but repeating that method chain multiple times is worse, IMO.

And then leverage that to clean up the block and add further comments on the flow. For example:

```ruby
# If a maximum limit is configured, we need to verify that the requested
# limit does not exceed the maximum.
if conjur_config.api_resource_list_limit_max.positive?
  # If no limit is given, default the limit to the configured maximum
  options[:limit] = conjur_config.api_resource_list_limit_max.to_s unless options[:limit]

  unless options[:limit].to_i <= conjur_config.api_resource_list_limit_max
    raise ApplicationController::UnprocessableEntity, \
          "'limit' parameter must not exceed #{conjur_config.api_resource_list_limit_max}"
  end
end
```

Reply:
done